src/test/java/com/spotify/docker/client/messages/HostConfigTest.java

```diff
@@ -65,6 +65,15 @@ public void testJsonOnFailure() throws Exception {
     assertThat(hostConfig.restartPolicy(), is(HostConfig.RestartPolicy.onFailure(5)));
   }
 
+  @Test
+  public void testIpcModeHostConfiguration() throws Exception {
+    HostConfig hostConfig = objectMapper
+            .readValue(fixture("fixtures/hostConfig/ipcModeHost.json"),
+                    HostConfig.class);
+    assertThat(hostConfig.ipcMode(), is("host"));
```

Comment:
This assertion only tests that the json deserializes properly into the object. While that is useful to know, there is a much better way to test that.

If you use this `HostConfig` as part of a `ContainerConfig`, then you use that to `createContainer`, you'll test that the `ipcMode` setting is accepted by the Docker Remote API. Then if you `inspectContainer` with the `id` of the container you just made, you can `assertThat` the `ipcMode` on the container you got back is equal to the `ipcMode` you set initially. That tests that the json (received from docker) deserializes properly, and that nothing happened to the value along the way.